### PR TITLE
Updated the code to implement the offline functionality for the stati…

### PR DESCRIPTION
**Pull request by ymaheshwari1**
…c pages (#1bevgv).

- First checking the LocalStorage for the static pages.
- Then, if we are online then hitting the url to get the latest data for the page and if offline the using TaskQueue to queue the request and hit the url when online.

https://app.clickup.com/t/1bevgv

### CODE REVIEW

File: store/action.ts [hunk 1]
```diff
@@ -29,19 +31,28 @@ export const actions: ActionTree<CmsState, any> = {
...
-      cmsJsonParser(resp).then(
-        (data) => {
-          cmsComponents = data
-          StorageManager.get('cmsStaticCollection').setItem(url.title, cmsComponents).catch((reason) => {
-            console.error(reason)
-          })
-          commit(types.GET_CMS_COMPONENTS, cmsComponents)
+    if(onlineHelper.isOnline) {
+      await fetch(`${PEREGRINE_CONFIG.endpoint}/${url.title}.data.json`, {
+        method: 'GET'
```
Comment (dixitdeepak) [inline store/action.ts:36]: I think you should use data-resolver as mentioned in https://docs.vuestorefront.io/guide/data-resolvers/introduction.html#how-to-create-a-data-resolver

Comment (ymaheshwari1) [reply store/action.ts]: Sir, used DataResolver for making the api calls.

File: data-resolver/CmsService.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { PEREGRINE_CONFIG } from '../helper/PeregrineConfig'
+import { DataResolver } from './types/DataResolver'
+import { TaskQueue } from '@vue-storefront/core/lib/sync'
+
```
Comment (mohammad-k8) [inline data-resolver/CmsService.ts:4]: I think we should name these functions as _fetchCmsData_ and _queueCmsData_. @dixitdeepak Please advise.

File: store/action.ts [hunk 3]
```diff
@@ -29,19 +31,22 @@ export const actions: ActionTree<CmsState, any> = {
...
-        }
-      )
-    })
+    if(onlineHelper.isOnline) {
+      await CmsService.fetchData(url, 'cmsstore/fetch')
+    } else {
+      await CmsService.fetchUpdatedData(url,'cmsstore/fetch')
+    }
+  },
+  async fetch({ commit }, resp) {
```
Comment (mohammad-k8) [inline store/action.ts:40]: We should name it as _loadCmsData_. @dixitdeepak Please advise.

Comment (ymaheshwari1) [reply store/action.ts]: Updated the method name as suggested

File: data-resolver/CmsService.ts (hunk 2 above)
Comment (ymaheshwari1) [reply data-resolver/CmsService.ts:4]: Updated the method name as suggested.

**Review by dixitdeepak — APPROVED** [review]

**Pull request merged**